src/main/java/com/microsoft/jenkins/azuread/AzureSecurityRealm.java

```diff
@@ -771,6 +861,10 @@ public FormValidation doVerifyConfiguration(@QueryParameter final String clientI
         }
     }
 
+    private static boolean isSecretEmpty(Secret secret) {
+        return secret == null || Secret.toString(secret).isEmpty();
```

Comment:
Secret.toString is null safe. (you might just be able to inline this)
```suggestion
        return Secret.toString(secret).isEmpty();
```